ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/KeyValueStorageProviderBuilder.java

```diff
@@ -61,6 +68,7 @@ public KeyValueStorageProvider build() {
     return new KeyValueStorageProvider(
         segments -> storageFactory.create(segments, commonConfiguration, metricsSystem),
         worldStatePreImageStorage,
-        (ObservableMetricsSystem) metricsSystem);
+        (ObservableMetricsSystem) metricsSystem,
+        deleteCodeInCodeHashStorageMode);
```

Comment:
We are leaking bonsai configuration concerns into the storage interfaces.  Can we not derive this in BonsaiWorldStateKeyValueStorage the same way we derive flatdb mode?  

We don't have (or shouldn't have) the synchronizer to make this db configuration for us the way flat db does.  I do wish we could push this into bonsai specific config rather than making it part of the storage provider interface.  

DataStorageFormat enum *could* have a defaulted member var for this config, and we could override it with a fluent function in the DataStorageFormat.  That somewhat breaks the concept of the enum being immutable.  

perhaps we just go with this and fast-follow with a PR that encapsulates storage config options.